server/utils/routes.js

```diff
@@ -2,18 +2,40 @@ module.exports = {
     getPathFor
 };
 
+function getPathFromAnswer({nextPath, data}) {
+    const decidingValue = data[nextPath.discriminator];
+    return nextPath[decidingValue];
+}
+
+function determinePathFromDecisions({decisions, data}) {
+    let path = null;
+    for (let pathConfig of decisions) {
+        const newPath = getPathFromAnswer({nextPath: pathConfig, data});
 
-function decidePath({decisionInfo, data}) {
-    const decidingValue = data[decisionInfo.discriminator];
-    return decisionInfo[decidingValue];
+        if (newPath) {
+            path = newPath;
+            break;
+        }
+    }
+
+    return path;
 }
 
-function getPathFor({formName, data, formConfig}) {
-    if (formConfig[formName].nextPath) {
-        return formConfig[formName].nextPath;
+function decidePath({nextPath, data}) {
+    if (nextPath.decisions && Array.isArray(nextPath.decisions)) {
+        const path = determinePathFromDecisions({decisions: nextPath.decisions, data});
+
+        return path || nextPath.path;
     }
-    if (formConfig[formName].nextPathDecision) {
-        return decidePath({decisionInfo: formConfig[formName].nextPathDecision, data});
+
+    if (nextPath.decisions) {
+        return getPathFromAnswer({nextPath: nextPath.decisions, data}) || nextPath.path;
     }
-    return null;
+
+
+    return nextPath.path;
```

Comment:
This can be tidied by initially checking !nextPath.decisions and returning nextPath.path. Then you don't need to keep checking for nextPath.decisions